File: stock-market-service/src/main/java/rs/edu/raf/model/StockHistoricalData.java
package rs.edu.raf.model;

import lombok.Getter;
import lombok.Setter;
import org.springframework.data.annotation.Id;
import org.springframework.data.redis.core.RedisHash;

import java.util.List;

@RedisHash("StockHistoricalData")
@Setter
@Getter
public class StockHistoricalData {
    @Id
    private String ticker;
    private List<StockData> historicalData;
}

File: stock-market-service-kotlin/src/main/kotlin/org/example/stockmarketservicekotlin/model/StockHistoricalData.kt
package org.example.stockmarketservicekotlin.model

import org.springframework.data.annotation.Id
import org.springframework.data.redis.core.RedisHash

@RedisHash("StockHistoricalData")
class StockHistoricalData(
    @Id
    val ticker: String,
    val historicalData: List<StockData>) {
}